core/metacling/src/TClingMethodInfo.h

```diff
@@ -54,13 +54,15 @@ class TClingTypeInfo;
 
 class TClingMethodInfo final : public TClingDeclInfo {
 private:
+   class UsingIterator;
    cling::Interpreter                          *fInterp; // Cling interpreter, we do *not* own.
    llvm::SmallVector<clang::DeclContext *, 2>   fContexts; // Set of DeclContext that we will iterate over.
    bool                                         fFirstTime; // Flag for first time incrementing iterator, cint semantics are weird.
    unsigned int                                 fContextIdx; // Index in fContexts of DeclContext we are iterating over.
    clang::DeclContext::decl_iterator            fIter; // Our iterator.
    std::string                                  fTitle; // The meta info for the method.
    const clang::FunctionDecl                   *fTemplateSpec; // an all-default-template-args function.
+   UsingIterator                               *fUsingIter; // for internal loop over using functions. [We own]
```

Comment:
```suggestion
   std::unique_ptr<UsingIterator>               fUsingIter; // for internal loop over using functions. [We own]
```
especially as `~TClingMethodInfo()` is already in the source file (and thus IIRC the class definition doesn't need `UsingIterator`)?